File: demo/vb/SetUpFixtureTests.vb
Imports NUnit.Framework

Namespace NUnitTestDemo.SetUpFixture

    <SetUpFixture>
    Public Class SetUpFixture
        Public Shared SetUpCount As Integer
        Public Shared TearDownCount As Integer

        <OneTimeSetUp>
        Public Sub BeforeTests()
            Assert.That(SetUpCount, Iz.EqualTo(0))
            SetUpCount = SetUpCount + 1
        End Sub

        <OneTimeTearDown>
        Public Sub AfterTests()
            Assert.That(TearDownCount, Iz.EqualTo(0))
            TearDownCount = TearDownCount + 1
        End Sub

    End Class

    <TestFixture, ExpectPass>
    Public Class TestFixture1

        <Test>
        Public Sub Test1()
            Assert.That(SetUpFixture.SetUpCount, Iz.EqualTo(1))
            Assert.That(SetUpFixture.TearDownCount, Iz.EqualTo(0))
        End Sub

    End Class

    <TestFixture, ExpectPass>
    Public Class TestFixture2

        <Test>
        Public Sub Test2()
            Assert.That(SetUpFixture.SetUpCount, Iz.EqualTo(1))
            Assert.That(SetUpFixture.TearDownCount, Iz.EqualTo(0))
        End Sub

    End Class

End Namespace

File: demo/cpp/SetUpFixtureTests.cpp
#include "ExpectedOutcomeAttributes.h"

using namespace System;
using namespace System::Collections::Generic;
using namespace System::Text;
using namespace NUnit::Framework;

namespace NUnitTestDemo
{
	namespace SetUpFixture
	{
		[SetUpFixture]
		public ref class SetUpFixture
		{
		public:
			static int SetUpCount;
			static int TearDownCount;

			[OneTimeSetUp]
			void BeforeTests()
			{
				Assert::That(SetUpCount, Is::EqualTo(0));
				SetUpCount++;
			}

			[OneTimeTearDown]
			void AfterTests()
			{
				Assert::That(TearDownCount, Is::EqualTo(0));
				TearDownCount++;
			}
		};

		[TestFixture, ExpectPass]
		public ref class TestFixture1
		{
		public:
			[Test]
			void Test1()
			{
				Assert::That(SetUpFixture::SetUpCount, Is::EqualTo(1));
				Assert::That(SetUpFixture::TearDownCount, Is::EqualTo(0));
			}
		};

		[TestFixture, ExpectPass]
		public ref class TestFixture2
		{
			[Test]
			void Test2()
			{
				Assert::That(SetUpFixture::SetUpCount, Is::EqualTo(1));
				Assert::That(SetUpFixture::TearDownCount, Is::EqualTo(0));
			}
		};
	}
}
